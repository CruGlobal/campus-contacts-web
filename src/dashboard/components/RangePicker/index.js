import styled from '@emotion/styled';
import React, { useState } from 'react';
import { withTheme } from 'emotion-theming';
import PropTypes from 'prop-types';
import moment from 'moment';
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import { DateRangePicker } from 'react-dates';

const Container = styled.div`
    display: flex;
    align-items: center;

    .DateRangePicker {
    }

    .DateRangePickerInput {
        display: flex;
    }

    .DateInput {
        width: 108px;
        height: 32px;
        text-align: center;
        display: flex;
        justify-content: center;
        align-items: center;
        background-color: #eceef2;
        text-align: center;
        border-radius: 0 16px 16px 0;
        overflow: hidden;
        cursor: pointer;

        &:first-child {
            border-radius: 16px 0 0 16px;
            margin-right: 1px;
        }

        &:hover {
            background-color: #e2e2e2;
        }

        input[type='text']:focus {
            background-color: #e2e2e2;
        }
    }

    .DateInput_input {
        height: 100%;
        font-size: 14px;
        line-height: 20px;
        padding: 0;
        background-color: transparent;
        border-color: transparent;
        border-radius: 0;
        text-align: center;
        color: ${({ theme }) => theme.colors.primary};
        cursor: pointer;
    }

    .CalendarDay__selected_span {
        background: ${({ theme }) => theme.colors.highlight};
        color: white;
        border: 1px solid #2fb0d8;
    }

    .CalendarDay__selected {
        background: ${({ theme }) => theme.colors.highlightDarker};
        color: white;
        border: 1px solid #2fb0d8;
    }

    .CalendarDay__selected:hover {
        background: ${({ theme }) => theme.colors.highlightDarker};
        color: white;
    }

    .CalendarDay__hovered_span:hover,
    .CalendarDay__hovered_span {
        background: ${({ theme }) => theme.colors.highlight};
        border: 1px solid #2fb0d8;
        color: white;
    }

    .DateRangePickerInput_arrow {
        display: none;
    }

    .DateInput_fang {
        top: 56px;
    }
`;

const CalendarIcon = styled.div`
    background-image: url(/src/dashboard/assets/icons/calendar.svg);
    width: 24px;
    height: 24px;
    margin-right: 12px;
`;

const RangePicker = ({ onDatesChange, startDate, endDate }) => {
    const [dates, setDates] = useState({ startDate, endDate });
    const [focus, setFocus] = useState();

    const datesChanged = ({ startDate, endDate }) => {
        setDates({ startDate, endDate });
        onDatesChange({ startDate, endDate });
    };

    return (
        <Container>
            <CalendarIcon />
            <DateRangePicker
                startDate={dates.startDate}
                startDateId="startDateId"
                endDate={dates.endDate}
                endDateId="endDateId"
                onDatesChange={datesChanged}
                focusedInput={focus}
                onFocusChange={focusedInput => {
                    setFocus(focusedInput);
                }}
                readOnly={true}
                noBorder={true}
                customArrowIcon={null}
                isOutsideRange={day => day.isAfter(moment.now())}
            />
        </Container>
    );
};

export default withTheme(RangePicker);

RangePicker.propTypes = {
    onDatesChange: PropTypes.func,
    startDate: PropTypes.object,
    endDate: PropTypes.object,
};
