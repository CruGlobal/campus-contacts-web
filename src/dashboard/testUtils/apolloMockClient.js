import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
