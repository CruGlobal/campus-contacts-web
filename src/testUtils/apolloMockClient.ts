import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
