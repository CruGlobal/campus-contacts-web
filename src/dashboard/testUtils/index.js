import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
