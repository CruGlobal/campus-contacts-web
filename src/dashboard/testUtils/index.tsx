import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
