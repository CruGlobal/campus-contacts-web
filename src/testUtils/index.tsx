import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
