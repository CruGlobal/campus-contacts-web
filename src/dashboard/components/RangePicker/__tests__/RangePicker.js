import React from 'react';

import { renderWithContext } from '../../../testUtils';
import RangePicker from '../';

describe('<RangePicker />', () => {
    it('should render properly', async () => {
        renderWithContext(<RangePicker />).snapshot();
    });
});
