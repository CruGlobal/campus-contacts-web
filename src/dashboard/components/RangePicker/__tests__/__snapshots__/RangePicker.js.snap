// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangePicker /> should render properly 1`] = `
<div>
  <div
    class="css-stddre"
  >
    <div
      class="css-dkr4m8"
    />
    <div
      class="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          class="DateRangePickerInput DateRangePickerInput_1"
        >
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-startDateId"
              aria-label="Start Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
              id="startDateId"
              name="startDateId"
              placeholder="Start Date"
              readonly=""
              type="text"
              value="07/05/2019"
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-startDateId"
            >
              Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-endDateId"
              aria-label="End Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
              id="endDateId"
              name="endDateId"
              placeholder="End Date"
              readonly=""
              type="text"
              value="07/12/2019"
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-endDateId"
            >
              Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
